### PR TITLE
upgrade to cypress 10

### PR DESCRIPTION
**Pull request by jpita**
upgrade to cypress 10

### CODE REVIEW

**Comment by jpita** [comment]
👋

**Comment by jpita** [comment]
`npm ERR! command sh -c cypress run --browser firefox --headless --record --key $CYPRESS_RECORD_KEY`

failing in CI due to lack of the variable, can someone check if the variable is present in circleCI?
I no longer have permissions

**Comment by hueitan** [comment]
Hi @jpita great to see you here. No idea what's happening to this permission, probably need some update from the CI configuration.

We still need the `package-lock.json`. Am i right the `cypress.json` is not necessary anymore because of having the new cypress config file?

**Comment by jpita** [comment]
Hi @hueitan !

miss you guys!!

package lock is in the files commuted right?

And yes, cypress.json is no longer needed in cypress 10

File: package.json [hunk 1]
```diff
@@ -16,9 +16,9 @@
     "deploy": "cross-env INSTRUMENTATION=1 RAND=1 npm run build && npm run adb:connect && ( npm run b2g:uninstall || echo ) && npm run b2g:install",
     "cypress:lint": "eslint cypress/",
     "cypress:fix-lint": "eslint cypress/ --fix",
-    "cypress:run": "cypress run --record --key $CYPRESS_RECORD_KEY",
-    "cypress:run:firefox": "cypress run --browser firefox --record --key $CYPRESS_RECORD_KEY",
-    "cypress:run:firefox:headless": "cypress run --browser firefox --headless --record --key $CYPRESS_RECORD_KEY",
```
Comment (hueitan) [inline package.json:21]: what's the purpose of $CYPRESS_RECORD_KEY in every run?

Comment (jpita) [reply package.json:21]: It send the recording to the cypress dashboard and attaches the results to it. 

It’s good to check for test analysis for flaky tests and reduce testing time. 

Not necessary for normal testing though, it’s a nice extra

**Pull request closed without merging**